src/otx/core/model/entity/visual_prompting.py

```diff
@@ -17,3 +24,35 @@ class OTXVisualPromptingModel(
 
     def __init__(self, num_classes: int = 0) -> None:
         super().__init__(num_classes=num_classes)
+
+
+class OTXZeroShotVisualPromptingModel(
+    OTXModel[ZeroShotVisualPromptingBatchDataEntity, ZeroShotVisualPromptingBatchPredEntity, T_OTXTileBatchDataEntity],
+):
+    """Base class for the zero-shot visual prompting models used in OTX."""
+
+    def __init__(self, num_classes: int = 0) -> None:
+        super().__init__(num_classes=num_classes)
+
+        self._register_load_state_dict_pre_hook(self.load_state_dict_pre_hook)
+
+    def state_dict(
+        self,
+        *args,
+        destination: dict[str, Any] | None = None,
+        prefix: str = "",
+        keep_vars: bool = False,
+    ) -> dict[str, Any] | None:
+        """Return state dictionary of model entity with reference features, masks, and used indices."""
+        super().state_dict(*args, destination=destination, prefix=prefix, keep_vars=keep_vars)
+
+        if isinstance(destination, dict):
+            # to save reference_info instead of reference_feats only
+            destination.pop(prefix + "model.reference_info.reference_feats")
+            destination.update({prefix + "model.reference_info": self.model.reference_info})
+        return destination
+
+    def load_state_dict_pre_hook(self, state_dict: dict[str, Any], prefix: str = "", *args, **kwargs) -> None:
+        """Load reference info manually."""
+        self.model.reference_info = state_dict.get(prefix + "model.reference_info", self.model.reference_info)
+        state_dict[prefix + "model.reference_info.reference_feats"] = self.model.reference_info["reference_feats"]
```

Comment:
Do we still need this even if we register them as `ParameterDict`? That means the native PyTorch handler for checkpoint can save and load them by itself.

Reply:
Because there are some components consisting of not only `Tensor` but also `list` and `dict`. As you said, native PyTorch handler can save and load `Tensor` automatically, but it seems not to be able to handle `list` and `dict`.

Reply:
I see, but do we need a lot of arguments here? Is `self` sufficient?

Reply:
`destination` is required to get children modules' state dict and update them to root `destination` at https://github.com/pytorch/pytorch/blob/main/torch/nn/modules/module.py#L1895.
![image](https://github.com/openvinotoolkit/training_extensions/assets/97221135/097d2546-553c-4937-99e4-604265765af6)

`prefix` is required to apply  lightning's prefix (currently `model.`).
`keep_vars` isn't used in this case, but I added it like the above others.

If you have better idea, please recommend :)

Reply:
Thanks for letting me know this. I didn't have background on how `nn.Module.state_dict()` working. However, I'm honestly feeling that we need more cleaning up for this. However, let's revisit afterwards.

Reply:
I agree with that.